src/util/address.rs

```diff
@@ -300,6 +300,27 @@ impl Address {
         self.address_type().is_some()
     }
 
+    /// Get an [Address] from an output script (scriptPubkey).
+    pub fn from_script(script: &script::Script, network: Network) -> Option<Address> {
+        Some(Address {
+            payload: if script.is_p2pkh() {
+                Payload::PubkeyHash(Hash::from_slice(&script.as_bytes()[3..23]).unwrap())
+            } else if script.is_p2sh() {
+                Payload::ScriptHash(Hash::from_slice(&script.as_bytes()[2..22]).unwrap())
+            } else if script.is_witness_program() {
+                // We can unwrap the u5 check and assume script length
+                // because [Script::is_witness_program] makes sure of this.
+                Payload::WitnessProgram {
+                    version: u5::try_from_u8(script.as_bytes()[0]).expect("checked before"),
```

Comment:
This is also incorrect - you need to decode the initial numeric push, not interpret it as a `u8`.